### DISCUSSION TITLE
Change Farmer Key and Pool Key of existing plot files

### DISCUSSION THREAD

**Discussion by FSys77** [Chia-Network/chia-blockchain#6534] [Show and tell]
Here someone claims that it is possible: https://www.youtube.com/watch?v=mlmtr6VEFOc

**Comment by rmica** [comment]
I read from the green paper:

... 
µ ← Sig.sign(sk, m) .
Chia uses signatures in the foliage (to chain foliage blocks and to bind them to
the trunk) and also in the trunk (so only the farmer can compute the challenge).
To avoid grinding attacks, the signatures used in the trunk must be unique, that
is for every pk (this includes maliciously generated public keys) and message m
there can be at most one accepting signature

∀pk, m, (Sig.verify(pk, m, µ) = accept) ∧ (Sig.verify(pk, m, µ0) = accept) ⇒ (µ = µ0) .
...

Looks like the problem will arise at farming (not plots create/check) time.
How can you sign with a new sk?


In solo farming, the plot id, the one the video says you can't change, is bound to the random plot key, and to the famer and pool secret keys, which for obvious reasons, don't get stored in the plot.
You can't sign m with a new secret key while farming, and verify it with (plot-id, m, sign).
Don't do that.
The farmer must sign with the secret key that the plot-id's public keys were derived from.

**Comment by mariano54** [comment]
Yes, this is not possible since your plot ID has to get updated, but the whole plot depends on the plot ID.
